cmd/uplink/cmd/cp.go

```diff
@@ -37,33 +36,19 @@ func init() {
 	progress = cpCmd.Flags().Bool("progress", true, "if true, show progress")
 }
 
-func cleanAbsPath(p string) string {
-	prefix := strings.HasSuffix(p, "/")
-	p = path.Join("/", p)
-	if !strings.HasSuffix(p, "/") && prefix {
-		p += "/"
-	}
-	return p
-}
-
 // upload uploads args[0] from local machine to s3 compatible object args[1]
-func upload(ctx context.Context, bs buckets.Store, srcFile string, destObj *url.URL) error {
-	var err error
-	if destObj.Scheme == "" {
-		return fmt.Errorf("Invalid destination")
-	}
-
-	destObj.Path = cleanAbsPath(destObj.Path)
+func upload(ctx context.Context, bs buckets.Store, srcFile fpath.FPath, destObj fpath.FPath) error {
```

Comment:
Let's add a guard here for checking for checking:
```
if !srcFile.IsLocal() || destObj.IsLocal() { return some error }
```

I know it's being checked outside as well, but I think it's nice to have clarity and future proofing in case it's called from somewhere else.

Reply:
Done.